### ISSUE TITLE
[Rule Request] Prefer comma-separated inheritance

### ISSUE THREAD

**Issue by marcelofabri** [realm/SwiftLint#3950]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### New rule request

1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

I don't think I've ever seen this mentioned in any style guide, but my gut feeling is that using commas is way more popular - especially since the `&` syntax came later in Swift's life.

2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// triggers
struct Size: Codable & Equatable {
  let width: Int
  let height: Int
}

// ok
struct Size: Codable, Equatable {
  let width: Int
  let height: Int
}
```

3. Should the rule be configurable, if so what parameters should be configurable?

Just severity

4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a rule as opt-in.

I think this is popular enough that it can be enabled by default.
